code/game/objects/items/teleportation.dm

```diff
@@ -179,7 +179,17 @@
 		var/area/computer_area = get_area(target)
 		if(!computer_area || (computer_area.area_flags & NOTELEPORT))
 			continue
-		if(computer.power_station?.teleporter_hub && computer.power_station.engaged)
+
+		if(!computer.power_station || !computer.power_station.teleporter_hub)
+			continue
+
+		if((computer.power_station.machine_stat & (NOPOWER|BROKEN|MAINT)) || computer.power_station.panel_open)
+			continue
+
+		if((computer.power_station.teleporter_hub.machine_stat & (NOPOWER|BROKEN|MAINT)) || computer.power_station.teleporter_hub.panel_open)
+			continue
```

Comment:
Life hack that apparently I'm the only one who knows about 

`computer.power_station.is_operational`
`computer.power_station.teleporter_hub.is_operational`

Reply:
This is a godsend!

Reply:
https://github.com/tgstation/tgstation/pull/88565/commits/d117bbe3fd85fa78faa8400f74ddf7ec889ee0de